File: Xamarin/Xamarin.Native/Xamarin.Native.Droid/Activities/MenuActivity.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using System;

namespace Xamarin.Native.Droid.Activities
{
    [Activity(Label = "Xamarin.Native.Droid", MainLauncher = true, Icon = "@drawable/icon")]
    public class MenuActivity : Activity
    {
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);
            SetContentView(Resource.Layout.Menu);
            Title = GetString(Resource.String.menu);

            Button performanceButton = FindViewById<Button>(Resource.Id.performanceButton);
            Button networkButton = FindViewById<Button>(Resource.Id.networkButton);
            Button locationButton = FindViewById<Button>(Resource.Id.locationButton);
            Button fileAccessButton = FindViewById<Button>(Resource.Id.fileAccessButton);

            performanceButton.Click += PerformanceButton_Click;
            networkButton.Click += NetworkButton_Click;
            locationButton.Click += LocationButton_Click;
            fileAccessButton.Click += FileAccessButton_Click;
        }

        private void FileAccessButton_Click(object sender, EventArgs e)
        {
            var i = new Intent(this, typeof(FileAccessTestActivity));
            StartActivity(i);
        }

        private void LocationButton_Click(object sender, EventArgs e)
        {
            var i = new Intent(this, typeof(LocationTestActivity));
            StartActivity(i);
        }

        private void NetworkButton_Click(object sender, EventArgs e)
        {
            var i = new Intent(this, typeof(NetworkTestActivity));
            StartActivity(i);
        }

        private void PerformanceButton_Click(object sender, EventArgs e)
        {
            var i = new Intent(this, typeof(PerformanceTestActivity));
            StartActivity(i);
        }
    }
}

File: Native.Android/app/src/main/java/com/pgrzmil/nativedroid/MenuActivity.java
package com.pgrzmil.nativedroid;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.app.Activity;

public class MenuActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setTitle(R.string.menu);
    }

    public void openPerformanceTestActivity(View view) {
        Intent i = new Intent(MenuActivity.this, PerformanceTestActivity.class);
        startActivity(i);
    }

    public void openNetworkTestActivity(View view) {
        Intent i = new Intent(MenuActivity.this, NetworkTestActivity.class);
        startActivity(i);
    }

    public void openLocationTestActivity(View view) {
        Intent i = new Intent(MenuActivity.this, LocationTestActivity.class);
        startActivity(i);
    }

    public void openFileAccessTestActivity(View view) {
        Intent i = new Intent(MenuActivity.this, FileAccessTestActivity.class);
        startActivity(i);
    }
}
